### DISCUSSION TITLE
How to get the variable values from other elements？

### DISCUSSION THREAD

**Discussion by stonebeta** [idaholab/moose#18150] [Q&A General] [answered]
Hello everyone,

I want to get the variable values from other elements in indicators. And those elements are within a specified distance from the current element. Now I can only get the variable values from the current element with ElementIndicator. 
Please help me, thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

That's not a very common need. If I may, what are you trying to do with that?

~~You could hack it and re-init at the neighbor element to get the variables there, but there may be a simpler way if you give us more details on what you want to do.~~
EDIT:
Actually it's not too bad for variable values. Use a pointer to the variable instead of the coupled interface to work with the variable, then use 
_var->getElementalValue(const Elem * neighbor, unsigned int index(<-for vectors/arrays) ) 

Guillaume

**Reply by stonebeta** [reply]
Thank you for your reply.  I guess the 'Elem * neighbor' is the element next to the current one.  Is there any way for a pointer to point to an arbitrary element ?

**Reply by GiudGiud** [reply]
You could have any element as an argument there, the problem is how do you retrieve the pointer in the kernel?


Neighbor is easy to get to from the attributes of the current element. You could retrieve an arbitrary element pointer from the mesh.
